app/scripts/detect-multiple-instances.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Sets up two-way communication between the
+ * mainline version of extension and Flask build
+ * in order to detect & warn if there are two different
+ * versions running simultaneously.
+ */
+
+import browser from 'webextension-polyfill';
+import {
+  METAMASK_PROD_BUILD_ID,
+  METAMASK_FLASK_BUILD_ID,
+} from '../../shared/constants/app';
+
+const MESSAGE_TEXT = 'isRunning';
+
+const showWarning = () =>
+  console.warn('Warning! You have multiple instances of MetaMask running!');
```

Comment:
It would be better to alert the user in this case, in the UI. A console warning seems like a good start though, and the user-facing alert could be handled in a later PR.

Reply:
afaik that is planned in a follow-up PR, @hivtushok wanted to keep the scope of this one smaller!